site/app/views/grading/ElectronicGraderView.php

```diff
@@ -806,6 +810,7 @@ public function detailsPage(Gradeable $gradeable, $graded_gradeables, $teamless_
             "team_gradeable_view_history" => $team_gradeable_view_history,
             "view_all" => $view_all,
             "anon_mode" => $anon_mode,
+            "anon_mode_cookie" => array_key_exists('anon_mode', $_COOKIE) ? $_COOKIE['anon_mode'] : 'off',
```

Comment:
Couldn't you use only anon_mode since that is set based off of your cookie already?

Reply:
The javascript needs access to the cookies value e.g const status = '{{ anon_mode_cookie }}. If I just passed anon_mode I wouldn't be able to update the cookie correctly.

Reply:
You could pass anon_mode into your JavaScript instead though.